execute `go test -race` failed

`go test -v -race ./...`

``` bash
$ go test -v -race ./...
=== RUN   TestGoPoolWithMutex
--- PASS: TestGoPoolWithMutex (0.11s)
=== RUN   TestGoPoolWithSpinLock
--- PASS: TestGoPoolWithSpinLock (0.11s)
=== RUN   TestGoPoolWithError
--- PASS: TestGoPoolWithError (0.11s)
=== RUN   TestGoPoolWithResult
--- PASS: TestGoPoolWithResult (0.11s)
=== RUN   TestGoPoolWithRetry
==================
WARNING: DATA RACE
Read at 0x00c00001c258 by goroutine 423:
  github.com/devchat-ai/gopool.TestGoPoolWithRetry()
      /workspaces/gopool/gopool_test.go:147 +0x284
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1629 +0x47

Previous write at 0x00c00001c258 by goroutine 523:
  github.com/devchat-ai/gopool.TestGoPoolWithRetry.func1()
      /workspaces/gopool/gopool_test.go:138 +0x64
  github.com/devchat-ai/gopool.(*worker).executeTaskWithoutTimeout()
      /workspaces/gopool/worker.go:78 +0xd1
  github.com/devchat-ai/gopool.(*worker).executeTask()
      /workspaces/gopool/worker.go:41 +0xc7
  github.com/devchat-ai/gopool.(*worker).start.func1()
      /workspaces/gopool/worker.go:26 +0xaa

Goroutine 423 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1629 +0x805
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2036 +0x8d
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2034 +0x87c
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1906 +0xb44
  main.main()
      _testmain.go:61 +0x2e9

Goroutine 523 (running) created at:
  github.com/devchat-ai/gopool.(*worker).start()
      /workspaces/gopool/worker.go:23 +0xf7
  github.com/devchat-ai/gopool.NewGoPool()
      /workspaces/gopool/gopool.go:75 +0x54f
  github.com/devchat-ai/gopool.TestGoPoolWithRetry()
      /workspaces/gopool/gopool_test.go:134 +0xfb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1576 +0x216
  testing.(*T).Run.func1()
      /usr/local/go/src/testing/testing.go:1629 +0x47
==================
    testing.go:1446: race detected during execution of test
--- FAIL: TestGoPoolWithRetry (0.10s)
=== NAME  
    testing.go:1446: race detected during execution of test
FAIL
FAIL    github.com/devchat-ai/gopool    0.558s
FAIL
```

--- a/gopool.go
+++ b/gopool.go
@@ -38,7 +38,7 @@ type goPool struct {
 	errorCallback func(error)
 	// adjustInterval is the interval to adjust the number of workers. Default is 1 second.
 	adjustInterval time.Duration
-	ctx    context.Context
+	ctx            context.Context
 	// cancel is used to cancel the context. It is called when Release() is called.
 	cancel context.CancelFunc
 }
@@ -57,8 +57,8 @@ func NewGoPool(maxWorkers int, opts ...Option) GoPool {
 		lock:           new(sync.Mutex),
 		timeout:        0,
 		adjustInterval: 1 * time.Second,
-		ctx:    ctx,
-		cancel: cancel,
+		ctx:            ctx,
+		cancel:         cancel,
 	}
 	// Apply options
 	for _, opt := range opts {

--- a/gopool_test.go
+++ b/gopool_test.go
@@ -3,6 +3,7 @@ package gopool
 import (
 	"errors"
 	"sync"
+	"sync/atomic"
 	"testing"
 	"time"
 
@@ -127,24 +128,24 @@ func TestGoPoolWithResult(t *testing.T) {
 }
 
 func TestGoPoolWithRetry(t *testing.T) {
-    var retryCount = 3
-    var taskError = errors.New("task error")
-    var taskRunCount = 0
+	var retryCount = int32(3)
+	var taskError = errors.New("task error")
+	var taskRunCount int32 = 0
 
-    pool := NewGoPool(100, WithRetryCount(retryCount))
-    defer pool.Release()
+	pool := NewGoPool(100, WithRetryCount(int(retryCount)))
+	defer pool.Release()
 
-    pool.AddTask(func() (interface{}, error) {
-        taskRunCount++
-        if taskRunCount <= retryCount {
-            return nil, taskError
-        }
-        return nil, nil
-    })
+	pool.AddTask(func() (interface{}, error) {
+		atomic.AddInt32(&taskRunCount, 1)
+		if taskRunCount <= retryCount {
+			return nil, taskError
+		}
+		return nil, nil
+	})
 
-    pool.Wait()
+	pool.Wait()
 
-    if taskRunCount != retryCount+1 {
-        t.Errorf("Expected task to run %v times, but it ran %v times", retryCount+1, taskRunCount)
-    }
+	if atomic.LoadInt32(&taskRunCount) != retryCount+1 {
+		t.Errorf("Expected task to run %v times, but it ran %v times", retryCount+1, taskRunCount)
+	}
 }

